src/main/c/Makefile

```diff
@@ -24,6 +24,7 @@ libbwa.$(LIB_EXT): $(JNI_BASE_NAME).o jnibwa.o bwa/libbwa.a
 
 bwa:
 	git clone https://github.com/lh3/bwa && cd bwa && git checkout $(BWA_MEM_COMMIT) && echo '#define BWA_COMMIT "'$(BWA_MEM_COMMIT)'"' > bwa_commit.h
+	sed -i .bak 's/\(LOBJS=.*\)/\1 $$(AOBJS)/g' bwa/Makefile
```

Comment:
I tried that once and bumped into a misterious seg-fault that was'nt able to clear.... I could give it a second try but considering that Apache2 branch is 90 commits behind bwa's master perhaps it could be worth to push for an update of that branch and then ```bwt_idx_build``` would be available.